### PR TITLE
bpf: Remove OUTPUT_SKB&HAS_KPROBE_MULTI macros

### PR DESCRIPTION
**Pull request by Asphaltt**
When call set_skb_btf() or set_shinfo_btf(), these functions are protected by constant CFG.

Then, while loading kprobe_pwru into kernel, if no --output-skb/--output-skb-shared-info, set_skb_btf()/set_shinfo_btf() won't be verified. Next, these unreachable code will be eliminated by verifier.

### CODE REVIEW

**Comment by jschwinger233** [comment]
Just for the record: 
1. 6.1 and bpf-next see `pwru --backend kprobe-multi` broken while `--backend kprobe` stil works.
2. 5.4 fails due to CO-RE issues on sk_buff. My guess is some members are deleted / renamed between 5.4 and 5.15.

File: internal/pwru/btf.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+/* Copyright Leon Hwang */
```
Comment (jschwinger233) [inline internal/pwru/btf.go:2]: Do you mind adding a line of `/* Copyright Authors of Cilium */` below, like https://github.com/cilium/pwru/blob/main/internal/pwru/types.go#L3 ?

File: bpf/features.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{
+	features.kprobe_happened = true;
+
+	/* Detect if bpf_snprintf_btf is available.
+	 * Since: c4d0bfb45068 ("bpf: Add bpf_snprintf_btf helper")
+	 */
+	features.has_snprintf_btf = bpf_core_enum_value_exists(enum bpf_func_id, BPF_FUNC_snprintf_btf);
+
+	return BPF_OK;
+}
```
Comment (jschwinger233) [inline bpf/features.c:24]: Can we use a similar approach to https://github.com/cilium/pwru/blob/v1.0.9/internal/pwru/utils.go#L138 for feature detection?

File: bpf/features.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Leon Hwang */
+
```
Comment (jschwinger233) [inline bpf/features.c:3]: Same Copyright issue.

**Review by jschwinger233 — COMMENTED** [review]
Always love to see code simplification! Left some comments to discuss.

File: bpf/features.c (hunk 2 above)
Comment (Asphaltt) [reply bpf/features.c:24]: I'll have a try.

**Review by brb — APPROVED** [review]
Thanks!

**Pull request merged**